### PR TITLE
OpenAPI: Allow one to use unix domain socket connection

### PR DESCRIPTION
**Pull request by Spredzy**
The following commit will allow one to use unix domain socket as part of
the pulp_url. Here what is a new call will look like

```
- name: Ensure Pulp is up and healthy
  pulp.squeezer.status:
    pulp_url: 'unix:/var/run/pulpcore-api/pulpcore-api.sock'
    username: 'admin'
    password: 'password'
```

### CODE REVIEW

File: plugins/module_utils/openapi.py [hunk 1]
```diff
@@ -42,9 +42,15 @@ def __init__(
         validate_certs=True,
         refresh_cache=False,
     ):
-        self.base_url = base_url
         self.doc_path = doc_path
 
+        if base_url.startswith("unix:"):
+            self.unix_socket = base_url.replace("unix:", "")
+            self.base_url = "http://localhost/"
```
Comment (mdellweg) [inline plugins/module_utils/openapi.py:49]: Interesting that this is needed on unix-sockets.

**Review by mdellweg — COMMENTED** [review]
Looks like a good change.
It is however not testable in the current test suite.
I think we can rely on this part not to change often. So if i do not get around testing this manually soon, i'll take your word that it works as designed.

File: plugins/module_utils/openapi.py (hunk 1 above)
Comment (Spredzy) [reply plugins/module_utils/openapi.py:49]: If you are interested https://github.com/ansible/ansible/pull/43560 is the PR that introduce this behavior and made it possible

**Comment by Spredzy** [comment]
```
[root@localhost ansible-tower-setup-3.8.0-0.git20201007.a170040]# cat toto.yml 
---
- hosts: localhost
  tasks:
    - name: Ensure Pulp is up and healthy
      pulp.squeezer.status:
        pulp_url: 'unix:/var/run/pulpcore-api/pulpcore-api.sock'
        username: 'admin'
        password: 'automationhub'

[root@localhost ansible-tower-setup-3.8.0-0.git20201007.a170040]# ansible-playbook -v toto.yml                                                                                                  
Using /home/vagrant/ansible-tower-setup-3.8.0-0.git20201007.a170040/ansible.cfg as config file                                                                                                  
[WARNING]: provided hosts list is empty, only localhost is available. Note that the implicit localhost does not match 'all'                                                                     
                                                                                                                                                                                                
PLAY [localhost] *******************************************************************************************************************************************************************************
                                                                                                                                                                                                
TASK [Gathering Facts] *************************************************************************************************************************************************************************
ok: [localhost]
                                                
TASK [Ensure Pulp is up and healthy] ***********************************************************************************************************************************************************
ok: [localhost] => changed=false         
  status:                  
    database_connection:                                                                        
      connected: true    
    online_content_apps:         
    - last_heartbeat: '2020-10-08T11:41:37.202054Z'                 
      name: 10951@localhost.localdomain                                                         
    - last_heartbeat: '2020-10-08T11:41:37.198701Z'
      name: 10950@localhost.localdomain
    online_workers:
    - last_heartbeat: '2020-10-08T11:41:31.131330Z'
      name: resource-manager
      pulp_created: '2020-10-08T08:01:56.459989Z'
      pulp_href: /pulp/api/v3/workers/c59c3b3e-f27e-47f3-82f6-44a3bd2b9de6/
    - last_heartbeat: '2020-10-08T11:41:31.278740Z'
      name: 11077@localhost.localdomain
      pulp_created: '2020-10-08T08:04:04.664781Z'
      pulp_href: /pulp/api/v3/workers/10805554-3d12-4dc4-a2a6-c79b991699d3/
    - last_heartbeat: '2020-10-08T11:41:31.535554Z'
      name: 11034@localhost.localdomain
      pulp_created: '2020-10-08T08:03:58.460118Z'
      pulp_href: /pulp/api/v3/workers/cbb7420e-dc75-4e1b-a7d9-1abb9c3e3885/
    redis_connection:
      connected: true
    storage:
      free: 38771466240
      total: 42927656960
      used: 4156190720
    versions:
    - component: pulpcore
      version: 3.7.1
    - component: pulp_container
      version: 2.1.0
    - component: pulp_ansible
      version: 0.4.1
    - component: galaxy_ng
      version: 4.2.0rc1

PLAY RECAP *************************************************************************************************************************************************************************************
localhost                  : ok=2    changed=0    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0
```

To provide you with more details.

@mdellweg also once this is merged could we get a `0.0.6` release with this patch. Its blocking us downstream.

File: plugins/module_utils/openapi.py (hunk 1 above)
Comment (mdellweg) [reply plugins/module_utils/openapi.py:49]: OK, I'll take your word for it. Seems like your folks will run kind of a nightly test for this. ;)

**Review by mdellweg — APPROVED** [review]

**Pull request merged**